src/common/streamwriter.h

```diff
@@ -34,20 +35,30 @@ class StreamWriter {
     void putInt64(int64_t val);
     void putCString(const std::string &str);
 
+    void write(char obj); // alias for putByte
+    void write(const std::string &obj);
+
+    /* only permits fundamental integral types or extended integral types */
+    template <typename T>
+    void write(T obj) {
+        swapBytesIfNotSystemEndianess(obj, _endianess);
+        _stream->write(reinterpret_cast<char*>(&obj), sizeof(T));
+    }
+
+    template<typename T>
+    StreamWriter &operator<<(T obj) {
+        write(obj);
+        return *this;
+    }
+
+    const std::shared_ptr<std::ostream> &getStream();
```

Comment:
Although slightly more efficient, returning a reference here breaks encapsulation, because now callers can change _stream directly. Should be: `std::shared_ptr<std::ostream> stream() const`

Reply:
Yeah I believe so